packages/eslint-plugin/test/__file_snapshots__/types/expect/expect-tests.ts.lint

```diff
@@ -0,0 +1,86 @@
+types/expect/expect-tests.ts
+   6:1  error  TypeScript expected type to be:
+  number
+got:
+  1234                                                                    @definitelytyped/expect
+  14:1  error  TypeScript expected type to be:
+  NotRightAtAll
+got:
+  1234                                                             @definitelytyped/expect
+  46:1  error  Cannot match a node to this assertion. If this is a multiline function call, ensure the assertion is on the line above  @definitelytyped/expect
+  50:1  error  TypeScript expected type to be:
+  NotMatched
+got:
+  any                                                                 @definitelytyped/expect
+
+✖ 4 problems (4 errors, 0 warnings)
+
+==== types/expect/expect-tests.ts ====
+
+    // eslint-disable-next-line @definitelytyped/no-relative-import-in-test
+    import * as expect from "./";
+    
+    
+    // $ExpectType number
+    expect.foo;
+    ~~~~~~~~~~
+!!! @definitelytyped/expect: TypeScript expected type to be:
+!!!                        :   number
+!!!                        : got:
+!!!                        :   1234
+    
+    
+    // $ExpectType 1234
+    expect.foo;
+    
+    
+    // $ExpectType NotRightAtAll
+    expect.foo;
+    ~~~~~~~~~~
+!!! @definitelytyped/expect: TypeScript expected type to be:
+!!!                        :   NotRightAtAll
+!!!                        : got:
+!!!                        :   1234
+    
+    
+    // $ExpectType string | number | undefined
+    expect.aUnion;
+    
+    // $ExpectType string | undefined | number
+    expect.aUnion;
+    
+    // $ExpectType number | string | undefined
+    expect.aUnion;
+    
+    // $ExpectType number | undefined | string
+    expect.aUnion;
+    
+    // $ExpectType undefined | string | number
+    expect.aUnion;
+    
+    // $ExpectType undefined | number | string
+    expect.aUnion;
+    
+    // $ExpectType any || undefined | number | string
+    expect.aUnion;
+    
+    // $ExpectType { prop1: "a" | "b" | "c"; prop2: readonly (string | number)[]; prop3: readonly (string | number)[]; }
+    expect.complicatedUnion(1, 2);
+    
+    // $ExpectType { prop1: "c" | "b" | "a"; prop2: ReadonlyArray<number | string>; prop3: ReadonlyArray<string | number>; }
+    expect.complicatedUnion(1, 2);
+    
+    
+    // $ExpectType NotMatched
+    // Whoops
+    ~~~~~~~~~
+!!! @definitelytyped/expect: Cannot match a node to this assertion. If this is a multiline function call, ensure the assertion is on the line above.
+    
+    
+    // $ExpectType NotMatched
+    
+    ~
+!!! @definitelytyped/expect: TypeScript expected type to be:
+!!!                        :   NotMatched
+!!!                        : got:
+!!!                        :   any
```

Comment:
It's not super clear to me whether or not this is a new bug or not; this should complain that it's unmatched. I'll have to test with the old code, but I did not change the logic so I assume that this is preexisting.

Reply:
Ah, it was finding `EndOfFileToken`. Fixed.